Thruster Volume changes + Overmap Tweaks + Charon changes

ghostsheet opened the pull request:
🆑
tweak: Overmap Fast speed is now reduced from 20 to 15, this applies to dodging meteors in shuttles at experienced/master levels.
tweak: Gas thrusters for overmap travel, will now calculate thrust and fuel consumption by volume. Lower volume limit from the engine control for more fuel efficiency.
tweak: Guppy's mass and max speed has been increased.
maptweak: Charon's atmospheric and electric compartment has be reworked to accommodate for the fuel consumption.
maptweak: There are now shuttle navpoints on the hanger deck. 
/🆑

Thrusters will now calculate thrust based on volume instead of a fixed number moles.
Thrust is now (roughly) calculate as root(Moles x molar mass x root(pressure)), where moles is taken from a volume of gas, instead of being arbitrary.
Basically not much has change front end, you'll still get a similar range of thrust in most circumstances
except there's more dynamic fuel consumption based on pressure/how dense the air is
And thrust is now also based on how dense the air is (molar volume). Which means you can go really fast if you freeze your gas lol.
This change would enable more complex mechanic to play with,
like varying fuel consumption and fuel efficiency and thrust calculations,
instead of just heat the gas until the pressure is high

While I was there, I fixed a small bug where the thrusters doesnt update its gas network, causing it to get stuck.

Changed Thrust limit to Volume limit on the engine console.

Increased both the Guppy's mass and max speed

Reduced Overmap fast speed to 15

Added some more shuttle Nav point on the hanger deck, reorganised the order shuttle navpoint, so it's less of a cluster fuck.

Add a co2 pressure tank to the charon for extra fuel reserve, they now have enough fuel on round start. (Reverted the fuel pump idea I had before)
Got #25591. Hopefully this way I can avoid the merge conflict

While I was mapping I also changed the guppy's door access to none, similar to the charon for ease of use from other departments. Cargo hatch is now locked to guppy access.

Reworked Charon
![dreammaker_6eFHBF1lHN](https://user-images.githubusercontent.com/43085828/57977644-fb59b780-7a3f-11e9-97bb-c2b33101143c.png)

ghostsheet committed:
Thruster Volume changes + Overmap Tweaks

Thrusters will now calculate thrust based on volume instead of moles.
Thrust is now (roughly) calculate as root(Moles x molar mass x root(pressure)), where moles is taken from a volume of gas, instead of being abitrary.
basically not much has change front end, except there's more fuel consumption and thrust based on molar volume.

Changed Guppy mass and max speed

Reduced fast speed to 15

Roland410 commented on code/__defines/overmap.dm:
```diff
@@ -3,5 +3,5 @@
-#define SHIP_SPEED_FAST  1/(5 SECONDS)
\ No newline at end of file
+#define SHIP_SPEED_SLOW  1/(40 SECONDS)
+#define SHIP_SPEED_FAST  3/(20 SECONDS)// 15 speed
```
As far as I'm aware no ship goes above 15, and most don't even go above 10 in normal circumstances.

BlueNexus commented:
This won't work properly, as no ship goes at such a speed. Could reduce it to 5?

ghostsheet committed:
Charon Map Update + Changes 2.0

Ask requested by Chinsky
the charon will now have enough internal storage gas to work upon spawn.

2.0----
Sprite fix on pressure tanks
Charon fuel port + fuel bay
Engine controls, changed thrust to volume

comma commented:
Charon needs to be able to go at least halway across the map and back without running out of fuel, without pre-flight muckery with additional canisters / cryochilling / goat sacrifices

ghostsheet commented:
Charon now has enough fuel for a round trip and some upgrades. But the way volume works is that it'll consume more fuel the more pressure you put in there, everyone can run out of fuel if they're not careful.

ghostsheet committed:
Shuttle waypoint + Fixes

Fixed a small engine bug not updating.
Add shuttle way point outside of the hanger deck
Fixed some engine ui
Added a camera to charon power compartment

comma commented:
Why 2 pumps chained like this? 
https://i.imgur.com/0VnZlog.png

ghostsheet commented:
the first pump on the top left was to make refueling from the canister to the pressure tank easier since you can fully empty the canister. the second one is to dump fuel into the engine which people might want to adjust.
if the first one seems redundant I can remove it.

comma commented:
Nah, was just unsure if accidental mapping thing.

PsiOmegaDelta commented on code/modules/atmospherics/pipes.dm:
```diff
@@ -1118,8 +1118,11 @@
 
-/obj/machinery/atmospherics/pipe/tank/Initialize()
+/obj/machinery/atmospherics/pipe/tank/air/New()
+	..()
```
Why do the two lines above here even exist?
The `icon_state = "air"` line could just be part of the type definition.

ghostsheet replied:
I think "air" and "air_map" is two slightly different sprite. Air is in the game version while air map is thing you see for mapping purposes
